### PR TITLE
Added more ciphers to CipherSuite. Refactored how all TLS config is created

### PR DESCRIPTION
**Pull request by AaronH88**
Based off of Shanes PR here: https://github.com/ansible/receptor/pull/718

### CODE REVIEW

File: pkg/backends/tcp.go [hunk 1]
```diff
@@ -227,8 +247,9 @@ func (ns *TCPSession) Close() error {
 // Command line
 // **************************************************************************
 
-// tcpListenerCfg is the cmdline configuration object for a TCP listener.
-type tcpListenerCfg struct {
+//TODO make these fields private
```
Comment (AaronH88) [inline pkg/backends/tcp.go:250]: Leaving this as TODO for now as this PR is already 200+ lines of code.

Will make a separate PR refactoring the scope of our fields

**Review by shanemcd — APPROVED** [review]
🚀
